app/views/layouts/_toast.html.erb

```diff
@@ -1,31 +1,25 @@
+
 <% if flash.any? %>
-  <div class="fixed top-4 left-[50%] translate-x-[-50%] w-full sm:w-1/2
-              flex flex-col gap-4 items-center
-              z-10
-              ">
-      <% flash.each do |type, text| %>
-        <%
-          toast_classes = case type
-                         when "notice"
-                           "bg-blue-100 text-blue-900 border-blue-400"
-                         when "alert"
-                           "bg-red-100 text-red-900 border-red-400"
-                         end
-        %>
-      <div
-          class="<%= toast_classes %> flex justify-between gap-4 rounded border items-baseline w-full pl-4"
-          data-controller="click-toggle"
-          data-click-toggle-flippable-class="hidden"
-          data-click-toggle-target="destination"
-      >
-          <p><%= text %></p>
-          <%= button_tag type: "button",
-                data: { action: "click-toggle#toggleClass" },
-                class: "p-4" do
-          %>
-            &times;
-          <% end %>
-        </div>
-      <% end %>
+  <div class="toast toast-top toast-end transition-transform duration-100 ease-in"
+        data-controller="transition"
+        data-transition-toggle-class="translate-x-full"
+        data-transition-target="me"
+        data-transition-after-timeout-value="3000"
+  >
```

Comment:
I'm not sure what using the daisy-ui classes here benefits. With the tooltips, you get a lot for a little -- the interactivity and calculations about when and how a tooltip shows up are free and hidden.

Here, daisy is just aliasing ordinary CSS where "toast-top" really means "fixed top-4" and so on. I'm particularly surprised that the daisy component doesn't have a close button, or automatically transition out. That would make it feel worth it to me.

Reply:
I think you're right, but I'm going to come back to that. I think there is a chance we re-think toasts further as part of larger design changes so I want to let things settle a bit more.

It probably would be nice to have an explicit X, but in this particular case it's actually hard to mouse over and click it before it disappears so I'll punt that to future consideration.